src/vmm/src/persist.rs

```diff
@@ -32,6 +43,139 @@ pub struct MicrovmState {
     pub device_states: DeviceStates,
 }
 
+/// Errors related to saving Microvm state.
+#[derive(Debug)]
+pub enum SaveMicrovmStateError {
+    /// Failed to save vCPU state.
+    SaveVcpuState(Option<vstate::Error>),
```

Comment:
nit: this is a bit weird, suggest
```suggestion
    InvalidVcpuState,
    SaveVcpuState(vstate::Error),
```

Reply:
Done.